plugin_build/README.md

```diff
@@ -0,0 +1,160 @@
+# GRPC Proxy How To
+
+**This is currently a very manual process due to how Golang's plugin mechanism works. We are working on a bakery-style
+docker image that will make this process more automated**
+
+*Edit* To make this easier, use the `tykio/bakery` container, it will generate a plugin file from your proto
+files without any additional steps.
+
+To add a gRPC PRoxy to Tyk, you will need a go build environment that is GRPC enabled, the first half of these instructions
+are lifted from the grpc-gateway project:
+
+## Step 1: Set up the base env
+
+First you need to install ProtocolBuffers 3.0.0-beta-3 or later.
+
+```sh
+mkdir tmp
+cd tmp
+git clone https://github.com/google/protobuf
+cd protobuf
+./autogen.sh
+./configure
+make
+make check
+sudo make install
+```
+
+Then, `go get -u` as usual the following packages:
+
+```sh
+go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
+go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
+go get -u github.com/golang/protobuf/protoc-gen-go
```

Comment:
I thought we vendored them?

Reply:
Not sure my last comment saved - if building the pb.go files manually you need to do this otherwise you can't actually generate the templates

Reply:
I thought we would rely on our vendored copies - otherwise plenty of things can go wrong, e.g. if those libs ever have any breaking changes

For example, we could tell people to use our `vendor/` as their `GOPATH`

Reply:
It doesn't affect these instructions, this command generates the plugins needed by `protoc` in the previous step to do the generating, so the stuff in our vendor dir won;t be part of the go bin path. 

They would need to change their GOPATH *after* when they actually build the plugin.